dataflow/run-inference/README.md

```diff
@@ -0,0 +1,147 @@
+gcloud pubsub topics create $MESSAGES_TOPIC
+gcloud pubsub topics create $RESPONSES_TOPIC
+```
+
+## Loading the `state_dict`
+
+LLMs can be **very large** models. Make sure the VMs you choose have enough memory to load them.
+Larger models generally give better results, but require more memory and can be slower to run on CPUs.
+You can add GPUs if you need faster responses.
+
+| Model                                   | Parameters | Memory needed |
+|-----------------------------------------|-----------:|--------------:|
+| [`google/flan-t5-small`][flan-t5-small] | 80M        | > 320 MB      |
+| [`google/flan-t5-base`][flan-t5-base]   | 250M       | > 1 GB        |
+| [`google/flan-t5-large`][flan-t5-large] | 780M       | > 3.2 GB      |
+| [`google/flan-t5-xl`][flan-t5-xl]       | 3B         | > 12 GB       |
+| [`google/flan-t5-xxl`][flan-t5-xxl]     | 11B        | > 44 GB       |
+| [`google/flan-ul2`][flan-ul2]           | 20B        | > 80 GB       |
+
+> 💡 Larger models require larger VMs to run.
+
+When running in Dataflow, Apache Beam's [`RunInference`](https://beam.apache.org/documentation/transforms/python/elementwise/runinference/) expects the model's  [`state_dict`](https://pytorch.org/tutorials/recipes/recipes/what_is_state_dict.html) to reside in Cloud Storage.
+
+Since LLMs can be so large, we want to save our `state_dict` as `float16` instead of the default `float32`.
+This means each parameter uses 16 bits instead of 32 bits, making the `state_dict` be half the size, which minimizes the time to load the model.
+However, converting the `state_dict` from `float32` to `float16` means our VM has to fit _both_ into memory.
+
+For smaller models, you can run it locally as long as you have enough memory to load the model and a fast internet connection to download the weights and upload them to Cloud Storage.
+Otherwise, we can launch a Vertex AI custom job to load it for us using a VM with the right size.
+The minimum (and default) disk size for Vertex AI is 100 GB, but some models might require a larger disk.
+
+Here's a table showing the minimum requirements to load a model's `state_dict`.
+
+| Model name             | Memory needed | Machine type    | VM Memory | VM Disk |
+|------------------------|---------------|-----------------|-----------|---------|
+| `google/flan-t5-small` | > 480 MB      | `e2-standard-4` | 16 GB      | 100 GB  |
+| `google/flan-t5-base`  | > 1.5 GB      | `e2-standard-4` | 16 GB      | 100 GB  |
+| `google/flan-t5-large` | > 4.8 GB      | `e2-standard-4` | 16 GB      | 100 GB  |
+| `google/flan-t5-xl`    | > 18 GB       | `e2-highmem-4`  | 32 GB     | 100 GB  |
+| `google/flan-t5-xxl`   | > 66 GB       | `e2-highmem-16` | 128 GB    | 100 GB  |
+| `google/flan-ul2`      | > 120 GB      | `e2-highmem-16` | 128 GB    | 150 GB  |
+
+```sh
+export MODEL_NAME="google/flan-t5-xl"
+export MACHINE_TYPE="e2-highmem-4"
+export DISK_SIZE_GB=100  # minimum is 100
+
+python load-state-dict.py vertex \
```

Comment:
could we change `load-state-dict.py` to `download_models.py`?

Reply:
Sure! I'll rename it